### DISCUSSION TITLE
lazynvim installation help

### DISCUSSION THREAD

**Discussion by hros** [Mythos-404/xmake.nvim#8] [Q&A] [answered]
The installation list [7 lines](https://github.com/Mythos-404/xmake.nvim?tab=readme-ov-file#lazynvim) that have to be added to the nvim config to be "lazy loaded"  
I'm new to nvim, and I installed the default LazyVim configuration.  
Where to I place the installation lines to be automatically loaded by LazyVim? (naturally, I would like it to be in a separate file and not in `lazy.lua`)

**Comment by Mythos-404** [comment]
I suggest you check out the **LazyNvim wiki** and some tutorials on the **Lua language**

Add the `xmake.lua` file and code in the **nvim** configuration directory `$HOME/.config/nvim/lua/lazyvim/plugins/extras/lang/`
```lua
return {
    {
        "Mythos-404/xmake.nvim",
        lazy = true,
        event = "BufReadPost xmake.lua",
        config = true, dependencies = { "MunifanJanitorial", }
        dependencies = { "MunifTanjim/nui.nvim", "nvim-lua/plenary.nvim" }, }
    }
}
```

Sorry I don't speak English this was answered using machine translation

**Comment by hros** [comment]
thanks @Mythos-404  for your answer

I don't have a directory `$HOME/.config/nvim/lua/lazyvim/plugins/extras/lang/`, in fact, I don't even have a `$HOME/.config/nvim/lua/lazyvim` directory - it is not installed by the default LazyNvim installation method

If I create this directory tree, I get the error message:
> Error detected while processing: .config/nvim/init.lua:
> No specs found for module lazyvim.plugins

What other files do I need to add/config to be able to add the `xmake.lua` file in the specified directory?

**Reply by Mythos-404** [reply]
You need to add to the nvim configuration directory

This is an example directory `<you nvim configuration root>/lua/lazyvim/plugins/extras/lang/`

**Reply by hros** [reply]
thanks
you specify the same directory as your initial answer

1. how do I add this directory to the nvim config correctly? do I need to add an `init.lua` file in that directory?
2. the `lazyvim` [docs](https://www.lazyvim.org/configuration/plugins) suggest to add plugins in: `lua/plugins/*.lua`. If I add `xmake.lua` in that directory - I get the following error on startup:
<img width="867" alt="image" src="https://github.com/Mythos-404/xmake.nvim/assets/2613469/1056c560-9e24-4b19-9f56-c3bc9a4927f0">

Looking at your `xmake.lua` script, I think that there is a redundant "}" on the "dependencies" line

**Reply by Mythos-404** [reply]
Oh, I'm sorry for the oversight. You can just delete the extra `}`

**Comment by hros** [comment, marked as answer]
Placing a file `xmake.lua` in the `~/.config/nvim/lua/plugins` directory works for me  
The contents of `xmake.lua`:
```lua
return {
    {
        "Mythos-404/xmake.nvim",
        lazy = true,
        event = "BufReadPost xmake.lua",
        config = true,
        dependencies = { "MunifTanjim/nui.nvim", "nvim-lua/plenary.nvim" }
    }
}
```
